app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperManager.kt

```diff
@@ -96,14 +99,21 @@ class WallpaperManager(
         }
     }
 
-    private fun filterExpiredRemoteWallpapers(wallpaper: Wallpaper): Boolean = when (wallpaper) {
+    private fun isWallpaperExpired(wallpaper: Wallpaper): Boolean = when (wallpaper) {
         is Wallpaper.Remote -> {
-            val notExpired = wallpaper.expirationDate?.let { Date().before(it) } ?: true
-            notExpired || wallpaper.name == settings.currentWallpaper
+            val expired = wallpaper.expirationDate?.let { Date().after(it) } ?: false
+            expired && wallpaper.name != settings.currentWallpaper
         }
-        else -> true
+        else -> false
     }
```

Comment:
The logic looks good, but I would rather avoid changing it now as this is something that we will very likely uplift directly to release (QA already battled tested the previous version) and we are close to prime time 😅 ,  I know we have a good set of tests, but I would avoid the risk.

Reply:
That makes sense. I'll revert the changes there